test/v2/controllers/requestParsers/validators/AmendUkPropertyAnnualSubmissionValidatorSpec.scala

```diff
@@ -0,0 +1,1278 @@
+              |    },
+              |    "adjustments": {
+              |      "lossBroughtForward": 343.34,
+              |      "privateUseAdjustment": 454.45,
+              |      "balancingCharge": 231.45,
+              |      "periodOfGraceAdjustment": true,
+              |      "businessPremisesRenovationAllowanceBalancingCharges": 567.67,
+              |      "nonResidentLandlord": true,
+              |      "rentARoom": {
+              |        "jointlyLet": true
+              |      }
+              |    }
+              |  },
+              |  "ukNonFhlProperty": {
+              |    "allowances": {
+              |      "annualInvestmentAllowance": 678.45,
+              |      "zeroEmissionGoodsVehicleAllowance": 456.34,
+              |      "businessPremisesRenovationAllowance": 573.45,
+              |      "otherCapitalAllowance": 452.34,
+              |      "costOfReplacingDomesticGoods": 567.34,
+              |      "electricChargePointAllowance": 454.34,
+              |      "structuredBuildingAllowance": [
+              |        {
+              |          "amount": 234.34,
+              |          "firstYear": {
+              |            "qualifyingDate": "2020-03-29",
+              |            "qualifyingAmountExpenditure": 3434.45
+              |          },
+              |          "building": {
+              |            "name": "Plaza",
+              |            "number": "1",
+              |            "postcode": "TF3 4EH"
+              |          }
+              |        }
+              |      ],
+              |      "enhancedStructuredBuildingAllowance": [
+              |        {
+              |          "amount": 234.45,
+              |          "firstYear": {
+              |            "qualifyingDate": "2020-05-29",
+              |            "qualifyingAmountExpenditure": 453.34
+              |          },
+              |          "building": {
+              |            "name": "Plaza 2",
+              |            "number": "2",
+              |            "postcode": "*"
+              |          }
+              |        }
+              |      ],
+              |      "zeroEmissionsCarAllowance": 454.34
+              |    },
+              |    "adjustments": {
+              |      "lossBroughtForward": 334.45,
+              |      "balancingCharge": 565.34,
+              |      "privateUseAdjustment": 533.54,
+              |      "businessPremisesRenovationAllowanceBalancingCharges": 563.34,
+              |      "nonResidentLandlord": true,
+              |      "rentARoom": {
+              |        "jointlyLet": true
+              |      }
+              |    }
+              |  }
+              |}
+              |""".stripMargin)
+        )) shouldBe List(StringFormatError.copy(paths = Some(Seq("/ukNonFhlProperty/allowances/enhancedStructuredBuildingAllowance/0/building/postcode"))))
+      }
+    }
+
+    "return ValueFormatError" when {
```

Comment:
`/ukNonFhlProperty/allowances/propertyIncomeAllowance` 
need a tests adding for this